src/Symfony/Bridge/Doctrine/Listener/DoctrineConnectionListener.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\Doctrine\Listener;
+
+use Doctrine\DBAL\Connection;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\RequestEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+class DoctrineConnectionListener implements EventSubscriberInterface
+{
+    private array $lastChecked = [];
+
+    public function __construct(private readonly ContainerInterface $container, private readonly array $timingByConnection)
+    {
+        foreach($this->timingByConnection as $params) {
+            $this->lastChecked[$params['name'] = 0];
+        }
+    }
+
+    public function onKernelRequest(RequestEvent $event): void
+    {
+        foreach ($this->timingByConnection as $params) {
+            $connection = $this->container->get($params['name']);
```

Comment:
I think we need to decouple the time check from the connection objects, so that we can skip loading anything from Doctrine if no ping is needed.
This means the listener should be able to know if the timeout arrived without using connection objects.
This means connection objects should populate a time registry when they do anything successful on the network.

Reply:
I'm not sure I understand. I need to associate a timing with each connection, right? Or perhaps I'm having difficulty understanding what you're suggesting

Reply:
Okay, I think I've got it, and I'm going to add a middleware. I'll work on it and push a commit. Thanks for your feedback!

Reply:
a WeakMap for example
we should be able to make a middleware that puts the current connection in the WeakMap at each network call, with the timestamp
and in the listener, we do a foreach on this map to know if you're closing (and unsetting) or not

Reply:
I'm not sure if I need a WeakMap. I could just have a simple array. The retrieved Connection is a decorator of Doctrine Dbal/Connection, and if I'm not mistaken, it closes on Dbal\Connection, right? https://github.com/doctrine/DoctrineBundle/pull/1739